language/move-vm/runtime/src/native_functions.rs

```diff
@@ -118,6 +123,23 @@ impl<'a, 'b> NativeContext<'a, 'b> {
             .debug_print_stack_trace(buf, self.resolver.loader())
     }
 
+    pub fn get_resource(
+        &mut self,
+        address: AccountAddress,
+        struct_tag: StructTag,
+    ) -> VMResult<&GlobalValue> {
+        let type_tag = TypeTag::Struct(Box::new(struct_tag));
+        let data_store = NativeContextDataStore {
+            data_store: self.data_store,
+        };
+
+        let type_ = self.resolver.loader().load_type(&type_tag, &data_store)?;
```

Comment:
Ok, the logic here is (wish the PR description would explain this instead of talking about "how long it take to introduce new instructions", which immediately lets me think this is a bad PR...):

- DataStore does not have exists_at but only get_resource
- get_resource requires a type
- in order to get the type, you need to pass an impl DataStore to the loader, dyn doesn't do

That's why you do all this here?

Reply:
would it help just change the StateView and resolver? from Stateview should be a quick change if I'm not mistaken, and if we can also propagate it through the resolver & DataStore and then access exists_at directly?
Happy to send smt out on the StateView side asap (and resolver too, I just never touched the Move-VM side yet).

Reply:
> would it help just change the StateView and resolver? from Stateview should be a quick change if I'm not mistaken, and if we can also propagate it through the resolver & DataStore and then access exists_at directly? Happy to send smt out on the StateView side asap (and resolver too, I just never touched the Move-VM side yet).

Check #868

Reply:
beautiful - thanks! I see - that should fix the problem here (and I guess the StateView isn't urgent and anyway being redesigned so we can expose exists directly now or later).

Completely orthogonal note but I also like exists interface e2e (whenever we just need exists) because validation for exists doesn't need to check the value, just the "exists" predicate.